src/platform/freertos/memory.cpp

```
@@ -57,6 +59,26 @@ static void ensure_pool_init() {
 
+void release_message_impl(someip::Message* msg) {
+    if (!msg) {
+        return;
+    }
+
+    msg->~Message();
+
+    auto* raw = reinterpret_cast<char*>(msg);
+    size_t const offset = static_cast<size_t>(raw - pool_buffer);
+    size_t const index = offset / sizeof(someip::Message);
+
```

Cppcheck (2.20.0)

[style] The function 'get_message_id' is never used.

(unusedFunction)